app/routes/tasks_routes.py

```diff
@@ -0,0 +1,134 @@
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+    
+
+@tasks_bp.route("",methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task.from_dict(request_body)
+    except:
+        return {"details": "Invalid data"} , 400
+    
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": {"id":new_task.id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": (new_task.completed_at != None)}}, 201
+
+@tasks_bp.route("",methods=["GET"])
+def read_all_tasks():
+
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    result = {
+        "task":{
+            "id": task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": (task.completed_at != None)}
+        }
+    if task.goal_id != None:
+        result["task"]["goal_id"] = task.goal_id
+    return result, 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task_to_update = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task_to_update.title = request_body["title"]
+    task_to_update.description = request_body["description"]
+
+    db.session.commit()
+
+    return jsonify({"task":task_to_update.to_dict()}), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return abort(make_response({"details":f"Task {task_id} \"{task.title}\" successfully deleted"}, 200))
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def update_task_complete_status(task_id):
+    task = validate_model(Task, task_id)
+    
+    if task.completed_at is None:
+        task.completed_at = datetime.utcnow()
+    
+        text_str = "Someone just completed the task " + task.title  
+        
+        url = 'https://slack.com/api/chat.postMessage'
+        payload = {
+                'token': os.environ.get("SLACK_API_TOKEN"),
+                'channel': '#task-notifications',
+                'text': text_str
+            }
+        
+        response = requests.post(url, data=payload)
```

Comment:
Consider making the `url` into a constant like `SLACK_URL` and/or adding it to a  constants file. Soon, you'll work on projects with multiple API calls that need to happen, and you'll be working with multiple urls, paylods, and api keys -- we could be more explicit by calling `payload` something like `SLACK_HEADERS` and `response` something like `slack_response`.